docs/crds/ssorule.md

```diff
@@ -0,0 +1,392 @@
+
+```yaml
+  oauthVsConfig:
+    cookieName: OAUTH_XYZ
+    cookieTimeout: 120
+    logoutURI: https://my-ssorule.test.com/oauth/logout
+    oauthSettings:
+    - appSettings:
+        clientID: my-client-id
+        clientSecret: my-oauth-secret
+        oidcConfig:
+          oidcEnable: true
+          profile: true
+          userinfo: true
+        scopes: ["scope1"]
+      authProfileRef: okta-oauth
+      resourceServer:
+        accessType: ACCESS_TOKEN_TYPE_OPAQUE
+        introspectionDataTimeout: 60
+        opaqueTokenParams:
+          serverID: my-server-id
+          serverSecret: my-oauth-secret
+    redirectURI: https://my-ssorule.test.com/oauth/callback
+    postLogoutRedirectURI: https://my-ssorule.test.com/oauth/postLogoutRedirectURI
+```
+The properties under the `oauthVsConfig` field are discussed in detail below.
+
+#### Express name for authorized session cookie
+
+The `cookieName` property can be used to express the name of HTTP cookie for an authorized session. If this field is not specified, NSX ALB will create the authorized session cookie with a random name.
+
+```yaml
+  cookieName: OAUTH_XYZ
+```
+
+#### Express timeout for authorized session cookie
+
+The `cookieTimeout` property can be used to specify the HTTP cookie timeout in **minutes** for an authorized session. It supports values from **1** to **1440** and defaults to **60**.
+
+```yaml
+  cookieTimeout: 120
+```
+
+#### Express Logout URI for OAuth
+
+The `logoutURI` property can be used to express the URI that will trigger OAuth logout.  
+
+```yaml
+  logoutURI: https://my-ssorule.test.com/oauth/logout
+```
+
+#### Express Application and IDP settings for OAuth/OIDC
+
+The `oauthSettings` property can be used to express the application specific OAuth configuration and Identity Provider (IDP) settings for OAuth/OIDC.  
+The application specific properties are specified as a list of `appSettings` properties.
+
+#### Express Client ID
+
+The `clientID` property can be used to express the client ID for the application. It is an application specific identifier registered with the authorization server or Identity Provider (IDP).
+
+```yaml
+  clientID: my-client-id
+```
+
+#### Express Client Secret
+
+The `clientSecret` field can be used to express the client secret for the application. It is an application specific identifier secret registered with the authorization server, or IDP. Since `clientSecret` is a sensetive field in NSX ALB, AKO requires it to be specified inside a Kubernetes Secret object. So, this `clientSecret` field should be the name of the Kubernetes secret object that specifies the actual client secret value (Base64 encoded) in the **clientSecret** data field.
+
+```yaml
+  clientSecret: my-oauth-secret
+```
+
+A sample Kubernetes secret object, with the actual value for client and secret (Base64 encoded) in the **clientSecret** data field is shown below.  
```

Comment:
Is it possible to create the secret in `avi-system` namespace and refer it in the SSORule? or are we looking for the secret in the namespace where the SSORule is present?